### PR TITLE
Revert aggressive page deletion - Flush cache and sleep

### LINKED ISSUE #1302: Revert aggresive page delete
Check 
- https://github.com/Automattic/wp-calypso/issues/81850
- p1695058112065739-slack-C02FMH4G8
- pdqkbG-38c-p2

Instead of deleting all pages, let's revert and delete only the Woo related ones

### LINKED ISSUE #1303: Purge cache after deleting woocommerce pages options
See 
- pdqkbG-38c-p2

TLDR; when deleting options and immediately getting them, object cache does not have enough time to invalidate

### PR DESCRIPTION
**Pull request by PanosSynetos**
### Changes proposed in this Pull Request:

Closes #1302 closes #1303 .
- #1302 
- #1303 

TLDR;

Instead of deleting all pages except for `'faq', 'contact-us', 'blog'`, we now delete only WooCommerce related pages.

We're extra careful, and check if the Woo Page we're about to delete is old. 

For example, if we have a simple site with a page called `shop` , when an Ecommerce plan is purchased, this page won't be overwritten. but instead we instruct WooCommerce to create a new page 

"Old" pages, are considered those that are older than 10 minutes.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [ ] Create a new Free site and add some pages.
- [ ] Get blog id from store admin
https://wordpress.com/wp-admin/network/admin.php?page=store-admin&action=blog_id_search&id=https%3A%2F%2FYOURTESTSITE.wordpress.com
- [ ] Go to https://mc.a8c.com/atomic/wpcom-dev-blogs/ 
- [ ] Enter the blog id, check `Transfers to Atomic dev server pool` and click `Add dev blog` 
- [ ] Upgrade to Business 
- [ ] Go to `Settings > Hosting management` and activate SFTP/SSH access (keep the credentials handy for the SSH and SCP commands further down)
- [ ] On your laptop, checkout branch `fix/1302-1303-revert-page-deletion-fix-cache` 
- [ ] CD into the local repo and `scp ./includes/class-wc-calypso-bridge-setup.php USERNAME@sftp.wp.com:/srv/htdocs/wp-content/mu-plugins/wpcomsh/vendor/automattic/wc-calypso-bridge/includes` to your WOA DEV site
- [ ] SSH into the new site and `tail -f -n1000 /tmp/php-errors` to see the logfile (keep it open)
- [ ] Click on `Upgrades` and upgrade to Ecommerce plan
- [ ] Check site that the pages created on the free site are not deleted
- [ ] Check that you have all WooCommerce-related pages (shop, cart, checkout, my-account, refund-returns)
- [ ] Check that there are no duplicate pages

You would see something similar to in your terminal - Please copy/paste it in your comments when you wrap up the test.
```
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8358) woocommerce_create_pages: INITIALIZED
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8361) woocommerce_create_pages: START TRANSACTION
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8365) woocommerce_create_pages: DELETING WOOCOMMERCE PAGES
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8443) woocommerce_create_pages: deleted page shop
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8496) woocommerce_create_pages: deleted page cart
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8547) woocommerce_create_pages: deleted page checkout
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8598) woocommerce_create_pages: deleted page my-account
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8645) woocommerce_create_pages: deleted page refund_returns
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8649) woocommerce_create_pages: DELETING WOOCOMMERCE PAGE OPTIONS
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8697) woocommerce_create_pages: deleted option woocommerce_shop_page_id : 13
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8704) woocommerce_create_pages: deleted option woocommerce_cart_page_id : 14
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.871) woocommerce_create_pages: deleted option woocommerce_checkout_page_id : 15
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8716) woocommerce_create_pages: deleted option woocommerce_myaccount_page_id : 16
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8764) woocommerce_create_pages: deleted option woocommerce_refund_returns_page_id : 17
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8764) woocommerce_create_pages: DELETING HEADSTART PAGES
[21-Sep-2023 09:12:11 UTC] WooExpress: Operation: (1695287531.8789) woocommerce_create_pages: FLUSH CACHE AND SLEEP
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.3796) woocommerce_create_pages: CREATING PAGES
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.3848) woocommerce_create_pages: woocommerce_create_pages filter - updated content to use cart/checkout blocks
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.388) woocommerce_create_pages: woocommerce_create_pages filter - pages:shop, cart, checkout, myaccount, refund_returns
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.3886) woocommerce_create_pages: woocommerce_create_page_id force create slug: shop
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.399) woocommerce_create_pages: woocommerce_page_created action - id: 18, slug: shop
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4096) woocommerce_create_pages: woocommerce_create_page_id force create slug: cart
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.419) woocommerce_create_pages: woocommerce_page_created action - id: 19, slug: cart
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4295) woocommerce_create_pages: woocommerce_create_page_id force create slug: checkout
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4378) woocommerce_create_pages: woocommerce_page_created action - id: 20, slug: checkout
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4481) woocommerce_create_pages: woocommerce_create_page_id force create slug: my-account
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4546) woocommerce_create_pages: woocommerce_page_created action - id: 21, slug: my-account
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4649) woocommerce_create_pages: woocommerce_create_page_id force create slug: refund_returns
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4742) woocommerce_create_pages: woocommerce_page_created action - id: 22, slug: refund_returns
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4836) woocommerce_create_pages: finished WC_Install::create_pages
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4959) woocommerce_create_pages: created menu items
[21-Sep-2023 09:12:12 UTC] WooExpress: Operation: (1695287532.4969) woocommerce_create_pages: COMMIT and cache deleted
```

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by markbiek — APPROVED** [review]
I don't think I can offer much about the specific implementation but it's working as expected!

```
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8333) woocommerce_create_pages: INITIALIZED
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8335) woocommerce_create_pages: START TRANSACTION
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8341) woocommerce_create_pages: DELETING WOOCOMMERCE PAGES
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8421) woocommerce_create_pages: deleted page shop
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8472) woocommerce_create_pages: deleted page cart
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.852) woocommerce_create_pages: deleted page checkout
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8568) woocommerce_create_pages: deleted page my-account
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8617) woocommerce_create_pages: deleted page refund_returns
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8621) woocommerce_create_pages: DELETING WOOCOMMERCE PAGE OPTIONS
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8671) woocommerce_create_pages: deleted option woocommerce_shop_page_id : 14
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8678) woocommerce_create_pages: deleted option woocommerce_cart_page_id : 15
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8684) woocommerce_create_pages: deleted option woocommerce_checkout_page_id : 16
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.869) woocommerce_create_pages: deleted option woocommerce_myaccount_page_id : 17
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8738) woocommerce_create_pages: deleted option woocommerce_refund_returns_page_id : 18
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8738) woocommerce_create_pages: DELETING HEADSTART PAGES
[21-Sep-2023 12:01:06 UTC] WooExpress: Operation: (1695297666.8763) woocommerce_create_pages: FLUSH CACHE AND SLEEP
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.377) woocommerce_create_pages: CREATING PAGES
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.3786) woocommerce_create_pages: woocommerce_create_pages filter - updated content to use cart/checkout blocks
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.3818) woocommerce_create_pages: woocommerce_create_pages filter - pages:shop, cart, checkout, myaccount, refund_returns
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.3824) woocommerce_create_pages: woocommerce_create_page_id force create slug: shop
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.393) woocommerce_create_pages: woocommerce_page_created action - id: 19, slug: shop
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.394) woocommerce_create_pages: woocommerce_create_page_id force create slug: cart
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4034) woocommerce_create_pages: woocommerce_page_created action - id: 20, slug: cart
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4135) woocommerce_create_pages: woocommerce_create_page_id force create slug: checkout
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4223) woocommerce_create_pages: woocommerce_page_created action - id: 21, slug: checkout
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4325) woocommerce_create_pages: woocommerce_create_page_id force create slug: my-account
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4395) woocommerce_create_pages: woocommerce_page_created action - id: 22, slug: my-account
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4497) woocommerce_create_pages: woocommerce_create_page_id force create slug: refund_returns
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4591) woocommerce_create_pages: woocommerce_page_created action - id: 23, slug: refund_returns
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4688) woocommerce_create_pages: finished WC_Install::create_pages
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4817) woocommerce_create_pages: created menu items
[21-Sep-2023 12:01:07 UTC] WooExpress: Operation: (1695297667.4827) woocommerce_create_pages: COMMIT and cache deleted
```

File: includes/class-wc-calypso-bridge-setup.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
  *
  * @package WC_Calypso_Bridge/Classes
  * @since   1.0.0
- * @version 2.2.13
+ * @version x.x.x
```
Comment (markbiek) [inline includes/class-wc-calypso-bridge-setup.php:7]: Is this correct or does it need to be an actual version number? I'm not super familiar with how the versioning works.

Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-setup.php:7]: Yes, we bump the `x.x.x`s when we're about to release the version, as we don't know in advance in which version this PR will make it.

File: includes/class-wc-calypso-bridge-setup.php [hunk 2]
```diff
@@ -680,6 +686,46 @@ public function prevent_redirects_on_activation( $location, $status ) {
...
+
+			$page_gmt_ts = get_post_time( 'U', true, $page['ID'] );
+			// draft pages don't have a post_date_gmt, so we need to calculate it.
+			if ( false === $page_gmt_ts ) {
+				$page_gmt_ts = get_gmt_from_date( $page['post_date'], 'U' );
+			}
+			$current_time_gmt_ts = current_time( 'U', true );
+			$diff_ts             = $current_time_gmt_ts - $page_gmt_ts;
+
+			if ( $diff_ts > 10 * MINUTE_IN_SECONDS ) {
```
Comment (daledupreez) [inline includes/class-wc-calypso-bridge-setup.php:714]: Do we want to ignore old Headstart pages as well? Note that we can look at the `_hs_old_id` `post_meta` field if we do want to delete Headstart pages in all cases.

File: includes/class-wc-calypso-bridge-setup.php [hunk 3]
```diff
@@ -680,6 +686,46 @@ public function prevent_redirects_on_activation( $location, $status ) {
...
+	 *
+	 * @since x.x.x
+	 *
+	 * @return void
+	 */
+	private function delete_page_by_slug( $slug, $operation ) {
+
+		$page = get_page_by_path( $slug, ARRAY_A );
+
+		if ( is_array( $page ) && isset( $page['ID'] ) ) {
```
Comment (daledupreez) [inline includes/class-wc-calypso-bridge-setup.php:704]: Totally minor, non-blocking suggestion: this might be a bit more readable as an early return.
```suggestion
		if ( ! is_array( $page ) || ! isset( $page['ID'] ) ) {
			return;
		}
```

File: includes/class-wc-calypso-bridge-setup.php [hunk 4]
```diff
@@ -680,6 +686,46 @@ public function prevent_redirects_on_activation( $location, $status ) {
 	private function write_to_log( $operation, $message ) {
 		error_log(  'WooExpress: Operation: (' . microtime( true ) . ') ' . $operation . ': ' . print_r( $message, 1 ) );
```
[the remark this answers is not in the document]
Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-setup.php:687]: Thanks for the suggestion - I'll apply it :)

File: includes/class-wc-calypso-bridge-setup.php (hunk 3 above)
Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-setup.php:705]: I'm all in for readability and bail out early! Thanks!

File: includes/class-wc-calypso-bridge-setup.php (hunk 2 above)
Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-setup.php:715]: Same principle as in my previous comment; 

If headstart pages are old, we don't know if they have been modified for whatever reason. So if they are old, we keep them. 

What do you think?

Comment (daledupreez) [reply includes/class-wc-calypso-bridge-setup.php]: That makes sense -- once they're too old, we don't want to make any assumptions about those pages.

File: includes/class-wc-calypso-bridge-setup.php [hunk 5]
```diff
@@ -224,7 +224,7 @@ public function woocommerce_create_pages_callback() {
 			$this->write_to_log( $operation, 'INITIALIZED' );
 
 			// Set the operation as completed if the store is active for more than 10 minutes.
```
Comment (daledupreez) [inline includes/class-wc-calypso-bridge-setup.php:226]: Nit: the comment says 10 minutes, but the code is now set to 5 minutes.
```suggestion
			// Set the operation as completed if the store is active for more than 5 minutes.
```

File: includes/class-wc-calypso-bridge-setup.php [hunk 6]
```diff
@@ -684,47 +684,50 @@ public function prevent_redirects_on_activation( $location, $status ) {
...
 	 *
 	 * @since x.x.x
 	 *
+	 * @param string $operation Operation.
+	 *
+	 * @param string $slug Slug.
 	 * @return void
 	 */
-	private function delete_page_by_slug( $slug, $operation ) {
+	private function maybe_delete_page_by_slug( $slug, $operation ) {
```
Comment (daledupreez) [inline includes/class-wc-calypso-bridge-setup.php:701]: Good call on using `maybe_` as the prefix!

**Review by daledupreez — APPROVED** [review]
Changes look good to me, though I have one minor note about a stale inline comment.

#### Screenshots

Screenshot 2023-09-26 at 13 24 22
Screenshot 2023-09-26 at 13 24 31

#### Log output

```
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.6997) INITIALIZED
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7) START TRANSACTION
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7004) DELETING WOOCOMMERCE PAGES 
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7086) deleted page shop
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7141) deleted page cart
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7196) deleted page checkout
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.725) deleted page my-account
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7307) deleted page refund_returns
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7311) DELETING WOOCOMMERCE PAGE OPTIONS 
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7383) deleted option woocommerce_shop_page_id : 30
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7392) deleted option woocommerce_cart_page_id : 31
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.74) deleted option woocommerce_checkout_page_id : 32
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7408) deleted option woocommerce_myaccount_page_id : 33
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7476) deleted option woocommerce_refund_returns_page_id : 34
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7476) DELETING HEADSTART PAGES 
[26-Sep-2023 11:21:57 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727317.7502) FLUSH CACHE AND SLEEP 
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.251) CREATING PAGES 
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.2584) woocommerce_create_pages filter - updated content to use cart/checkout blocks
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.2614) woocommerce_create_pages filter - pages:shop, cart, checkout, myaccount, refund_returns
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.262) woocommerce_create_page_id force create slug: shop
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.2722) woocommerce_page_created action - id: 35, slug: shop
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.2866) woocommerce_create_page_id force create slug: cart
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.296) woocommerce_page_created action - id: 36, slug: cart
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3098) woocommerce_create_page_id force create slug: checkout
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3178) woocommerce_page_created action - id: 37, slug: checkout
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3312) woocommerce_create_page_id force create slug: my-account
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3375) woocommerce_page_created action - id: 38, slug: my-account
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3511) woocommerce_create_page_id force create slug: refund_returns
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3611) woocommerce_page_created action - id: 39, slug: refund_returns
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3742) finished WC_Install::create_pages
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3864) created menu items
[26-Sep-2023 11:21:58 UTC] WooExpress: Operation: woocommerce_create_pages: (1695727318.3874) COMMIT and cache deleted
```
